File: CSharp/mars-rover/Plateau.cs
using mars_rover.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace mars_rover
{   
    public static class MainPlateauClass
    {
        public static bool IsMoveAvailable(this Plateau plateau, Position position)
        {
            return plateau.Min_Width <= position.x && position.x <= plateau.Width && plateau.Min_Height <= position.y && position.y <= plateau.Height;
        }
    }
}

File: Python/marsrover/plateau.py
class Plateau(object):
    MIN_WIDTH = 0
    MIN_HEIGHT = 0

    def __init__(self, width, height, min_width=0, min_height=0):
        self.width = width
        self.height = height
        self.MIN_WIDTH = min_width
        self.MIN_HEIGHT = min_height

    def move_available(self, position):
        """

        :param Position position:
        :return:
        """
        return self.MIN_WIDTH <= position.x <= self.width and self.MIN_HEIGHT <= position.y <= self.height
